service/src/test/java/com/ge/predix/acs/privilege/management/dao/GraphResourceRepositoryTest.java

```diff
@@ -141,8 +157,81 @@ public void testGetByZoneAndResourceIdentifier() {
         persistResource1toZone2();
         assertThat(IteratorUtils.count(this.graph.vertices()), equalTo(2L));
 
-        ResourceEntity resource = this.resourceRepository.getByZoneAndResourceIdentifier(TEST_ZONE_1, RESOURCE_1_ID);
-        assertThat(resource.getResourceIdentifier(), equalTo(RESOURCE_1_ID));
+        ResourceEntity resource = this.resourceRepository.getByZoneAndResourceIdentifier(TEST_ZONE_1,
+                RESOURCE_XFILE_1_ID);
+        assertThat(resource.getResourceIdentifier(), equalTo(RESOURCE_XFILE_1_ID));
+    }
+
+    @Test
+    public void testGetByZoneAndResourceIdentifier2LevelHierarchical() {
+        assertThat(IteratorUtils.count(this.graph.vertices()), equalTo(0L));
+        String resourceIdentifier = persist2LevelHierarchicalResource1toZone1().getResourceIdentifier();
+        assertThat(IteratorUtils.count(this.graph.vertices()), equalTo(2L));
+
+        ResourceEntity resource = this.resourceRepository.getByZoneAndResourceIdentifier(TEST_ZONE_1,
+                resourceIdentifier);
+        assertThat(resource.getAttributes().contains(ATTR_SITE_0), equalTo(true));
+        assertThat(resource.getAttributes().contains(ATTR_TYPE_1), equalTo(true));
+    }
+
+    @Test
+    public void testGetByZoneAndResourceIdentifier3LevelHierarchical() {
+        assertThat(IteratorUtils.count(this.graph.vertices()), equalTo(0L));
+        String resourceIdentifier = persist3LevelHierarchicalResource1toZone1().getResourceIdentifier();
+        assertThat(IteratorUtils.count(this.graph.vertices()), equalTo(3L));
+
+        ResourceEntity resource = this.resourceRepository.getByZoneAndResourceIdentifier(TEST_ZONE_1,
+                resourceIdentifier);
+        assertThat(resource.getAttributes().contains(ATTR_SITE_0), equalTo(true));
+        assertThat(resource.getAttributes().contains(ATTR_TYPE_1), equalTo(true));
+        assertThat(resource.getAttributes().contains(ATTR_CLASSIFICATION_0), equalTo(true));
```

Comment:
this test will be even more easier to follow, if the attributes being asserted were passed in to the utility method at line 180  , where the hierarchy is created.